apps/aefate/src/aefa_fate_op.erl

```diff
@@ -138,34 +141,75 @@ returnr(Arg0, EngineState) ->
     ES2 = aefa_fate:check_return_type(ES1),
     aefa_fate:pop_call_stack(ES2).
 
-
 call(Arg0, EngineState) ->
-    {Fun, ES0} = get_op_arg(Arg0, EngineState),
-    ES1 = aefa_fate:push_return_address(ES0),
+    ES1 = aefa_fate:push_return_address(EngineState),
+    call_t(Arg0, ES1).
+
+call_t(Arg0, EngineState) ->
+    {Fun, ES1} = get_op_arg(Arg0, EngineState),
     Signature = aefa_fate:get_function_signature(Fun, ES1),
-    {ok, ES2} = aefa_fate:check_signature_and_bind_args(Signature, ES1),
+    ok = aefa_fate:check_signature(Signature, ES1),
+    ES2 = aefa_fate:bind_args_from_signature(Signature, ES1),
     {jump, 0, aefa_fate:set_local_function(Fun, ES2)}.
 
-call_r(Arg0, Arg1, EngineState) ->
+call_r(Arg0, Arg1, Arg2, EngineState) ->
+    Sender = aefa_engine_state:current_contract(EngineState),
```

Comment:
Could be done in `call_r_common`?

Reply:
:thinking: Yes. It just needs to happen before the binding of new arguments.